prusti-specs/src/lib.rs

```diff
@@ -421,3 +421,37 @@ pub fn extern_spec(_attr: TokenStream, tokens:TokenStream) -> TokenStream {
         _ => { unimplemented!() }
     }
 }
+
+pub fn predicate(attr: TokenStream, tokens: TokenStream) -> TokenStream {
+    if !attr.is_empty() {
+        // TODO: what here? error? re-emit the previous attributes? not sure
+        //       what the semantics for that are supposed to be..
```

Comment:
Do you know an example that triggers this? In the other macros we just call the first parameter `_attr` and we never use it.

Reply:
turns out i misunderstood what this parameter is used for; just ignoring it seems like the best idea, given that's how the rest of the codebase deals with this, will change.

Reply:
`#[predicate(123)]` looks like an example, in which case raising a compilation error as you mention in the TODO sounds good. The `pure` and `trusted` macros don't do this check, but I think they should.

Reply:
To return a compilation error you can use `return syn::Error::new(span, "error message").to_compile_error();`. I hope you can obtain the span from `attr`, somehow.

Reply:
i'd be happy to address this, but would open a separate PR for that, what do you think?

Reply:
For `pure` and `trusted` yes.

Reply:
https://github.com/viperproject/prusti-dev/pull/391